tls/s2n_resume.c

```diff
@@ -280,7 +280,12 @@ static S2N_RESULT s2n_tls12_client_deserialize_session_state(struct s2n_connecti
 
 static S2N_RESULT s2n_validate_ticket_age(uint64_t current_time, uint64_t ticket_issue_time)
 {
-    RESULT_ENSURE(current_time >= ticket_issue_time, S2N_ERR_INVALID_SESSION_TICKET);
+    /* In the event of clock skew or a clock jump, we assume that the clock skew
+     * is less than one week and therefore the ticket is valid.
+     */
+    if (current_time < ticket_issue_time) {
+        return S2N_RESULT_OK;
+    }
```

Comment:
Instead of just accepting all future tickets, can we apply some sanity checks? You assume < one week, but nothing enforces that. I'd say the "margin of error" should either be set reasonably very low or should minimally be related to the length of the ticket lifetime. We don't want to accidentally double the ticket lifetime or something.

Reply:
tl;dr: I think capping it is fine, but I'd push for a full week of allowance (making things symmetric)

### reasoning

This change does not increase the age of tickets that we may accept, although it may increase the *average* age. 

Consider two quantities, `true_ticket_age` and `issuing_host_future_skew`.

Currently we accept the ticket of the following inequalities are satisfied
```
issuing_host_future_skew > true_ticket_age - ONE_WEEK
AND
issuing_host_future_skew < true_ticket_age
```

It's much easier to see graphed out [here](https://www.desmos.com/calculator/sqa0uyw5fi) where `x` is the the `true_ticket_age` and `y` is `issuing_host_future_skew`. (y -> positive: skew further in the future, y -> negative: skew further in the past)

This matches the intuition that the below example gives us
```
clock skew: 2 hours, ticket age: 1 hour -> assumed age -1 hour ❌ 
clock skew: 1 hour, ticket age: 2 hours -> assumed age 1 hour ✅ 
clock skew: 100_000 hours, ticket age: 100_001 hours -> assumed age 1 hour ✅ 
```

Now it's true that we would be increasing the number of scenarios where we could accept the old tickets
```
clock skew: 100_000 hours, ticket_age: 50_000 hours
currently -> ❌ 
proposed -> ✅ 
```

To cut down on this I agree that we could cap the future allowance, but I don't think it makes sense to be so stingy when we already accept such a wide range of clock skews/ages. I would propose just making the bounds symmetric, and just asserting that it is no farther in the future than 1 week.